utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dsl.lua

```diff
@@ -0,0 +1,98 @@
+--!/usr/bin/lua
+
+local function scrape()
+        local fs = require "nixio.fs"
+	local has_dsl = fs.access("/etc/init.d/dsl_control", "x")
+	if has_dsl then
+		local dsl_uptime = metric("dsl_uptime", "gauge")
+		local dsl_up = metric("dsl_up", "counter")
+		local dsl_data_rate_up = metric("dsl_data_rate_up", "gauge")
+		local dsl_data_rate_down = metric("dsl_data_rate_down", "gauge")
+		local dsl_max_data_rate_up = metric("dsl_max_data_rate_up", "gauge")
+		local dsl_max_data_rate_down = metric("dsl_max_data_rate_down", "gauge")
+		local dsl_errors_total = metric("dsl_errors_total", "counter")
+		local dsl_error_seconds_total = metric("dsl_error_seconds_total", "counter")
+		local dsl_line_attenuation = metric("dsl_line_attenuation_db", "gauge")
+		local dsl_signal_attenuation = metric("dsl_signal_attenuation_db", "gauge")
+		local dsl_snr = metric("dsl_signal_to_noise_margin_db", "gauge")
+		local dsl_aggregated_transmit_power = metric("dsl_aggregated_transmit_power_db", "gauge")
+
+                local sys  = require "luci.sys"
```

Comment:
You are doing `require luci.sys` just for `exec`, ie loading lots of unneeded code in memory every 15 sec
`exec` is only 3 line long: https://github.com/openwrt/luci/blob/master/modules/luci-base/luasrc/util.lua#L598
Also some people might not want to have luci installed